app/src/main/java/mozilla/lockbox/store/SecureStore.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.store
+
+import android.content.SharedPreferences
+import android.util.Base64
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.subjects.PublishSubject
+import mozilla.components.lib.dataprotect.Keystore
+import mozilla.components.service.fxa.OAuthInfo
+import mozilla.components.service.fxa.Profile
+import mozilla.lockbox.action.SecureAction
+import mozilla.lockbox.extensions.filterByType
+import mozilla.lockbox.flux.Dispatcher
+import java.nio.charset.StandardCharsets
+
+private const val KEYSTORE_LABEL = "lockbox-keystore"
+private const val OAUTH_KEY = "firefox-account-oauth-info"
+private const val PROFILE_KEY = "firefox-account-profile"
+
+private const val BASE_64_FLAGS = Base64.URL_SAFE or Base64.NO_PADDING or Base64.NO_PADDING
+
+class SecureStore(
+    private val dispatcher: Dispatcher = Dispatcher.shared,
+    private val keystore: Keystore = Keystore(KEYSTORE_LABEL)
+) {
+    companion object {
+        val shared = SecureStore()
+    }
+
+    private lateinit var prefs: SharedPreferences
+    internal val compositeDisposable = CompositeDisposable()
+
+    val oauthInfo: Observable<OAuthInfo?> = PublishSubject.create()
+    val profile: Observable<Profile?> = PublishSubject.create()
+
+    init {
+        this.dispatcher.register
+            .filterByType(SecureAction::class.java)
+            .subscribe {
+                when (it) {
+                    is SecureAction.OAuthInfo -> {
+                        // tbd: how to convert to & from String values for OAuthInfo...
```

Comment:
yeah yeah yeah, WIP (-:

This might provide some pointers for the `tbd`s: [samples/firefox-accounts](https://github.com/mozilla-mobile/android-components/blob/dbaf87110dbf5f7183ba067c086049f4b4ff5bb6/samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt#L137)

Reply:
thanks matt :)